flink-learning-common/src/test/java/com/zhisheng/common/utils/DateUtilTests.java

```diff
@@ -0,0 +1,71 @@
+package com.zhisheng.common.utils;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import static com.zhisheng.common.utils.DateUtil.*;
+import static org.junit.Assert.*;
+
+public class DateUtilTests {
+
+    @Test
+    public void testFormat() throws ParseException {
+        SimpleDateFormat myDate1 = new SimpleDateFormat("yyyy-MM-dd");
+        SimpleDateFormat myDate2 = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+        SimpleDateFormat myDate3 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        SimpleDateFormat myDate4 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.0");
+
+        assertEquals(myDate3.parse(myDate3.format(new Date())).getTime(), format(new Date()));
+
+        assertEquals(myDate1.format(new Date()), format(new Date().getTime(), DateUtil.YYYY_MM_DD));
+        assertEquals(myDate1.format(new Date()), format(new Date(), DateTimeZone.getDefault(), DateUtil.YYYY_MM_DD));
+
+        long testDate1 = myDate1.parse(myDate1.format(new Date())).getTime();
+        long testDate2 = myDate2.parse(myDate1.format(new Date())).getTime();
+        long testDate3 = myDate3.parse(myDate1.format(new Date())).getTime();
+        long testDate4 = myDate4.parse(myDate1.format(new Date())).getTime();
+
+        assertEquals(testDate1, format(myDate1.format(new Date()), YYYY_MM_DD));
+        assertEquals(testDate2, format(myDate2.format(new Date()), YYYY_MM_DD_HH_MM));
+        assertEquals(testDate3, format(myDate3.format(new Date()), YYYY_MM_DD_HH_MM_SS));
+        assertEquals(testDate4, format(myDate4.format(new Date()), YYYY_MM_DD_HH_MM_SS_0));
+    }
+
+    @Test
+    public void testIsValidDate() {
+        assertTrue(isValidDate("2019-05-01", YYYY_MM_DD));
+
+        assertFalse(isValidDate("01-05-2019", YYYY_MM_DD));
+    }
+
+    @Test
+    public void testToDate() throws ParseException {
+        SimpleDateFormat myDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+        myDate.setTimeZone(TimeZone.getTimeZone("UTC"));
+        assertEquals(myDate.parse("2019-04-30T23:00:00"), toDate("2019-05-01", YYYY_MM_DD));
```

Comment:
You could also initialise the `expected` value using the `Date` constructor `new Date(long millis)`, to remove the uncertainty about `SimpleDateFormat`.

Reply:
Done